testdata/p4_16_samples/issue2800c.p4

```diff
@@ -0,0 +1,36 @@
+#include <core.p4>
+
+header H {
+    bit<16> a;
+    int<16> b;
+}
+
+
+struct metadata {
+}
+
+struct Headers {
+    H h;
+}
+
+parser p(packet_in packet, out Headers hdr) {
+    state start {
+        packet.extract(hdr.h);
+        transition select(hdr.h.a, hdr.h.b) {
+            (0 .. 7, -5 .. 4): parse;
```

Comment:
How about a test that has 2 ranges containing 0?

Reply:
Sure, I have added one more test with two signed fields (not both of them are going over 0 since I realized I did not have any test for purely negative range. I think the "going over zero" case is covered sufficiently and this test mainly covers the missing "multiple fields are signed" case.